app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Any, Dict, Tuple
+
+
+def cache(func: callable) -> callable:
+    cached_results: Dict[Tuple[str, Tuple, frozenset], Any] = {}
+
+    def wrapper(*args: Any, **kwargs: Any) -> Any:
+        key: Tuple[str, Tuple, frozenset] = (func.__name__, args,
```

Comment:
No need to add a function name to the key because the cache dict is unique for each function. It is created while decorating.